cfme/cloud/provider.py

```diff
@@ -230,6 +258,8 @@ def _form_mapping(self, create=None, **kwargs):
             })
         return data_dict
 
+OpenStackProvider.add_type_map(OpenStackProvider)
```

Comment:
at first glance i think something like a decorator, like `@Provider.add_to_type_map` might be a good readability enhancement